### PR TITLE
[WIN32K:ENG] EngAlphaBlend(): Release EnterLeaveSource on error

### PR DESCRIPTION
**Pull request by SergeGautherie**
Part of
JIRA issue: [CORE-11156](https://jira.reactos.org/browse/CORE-11156)

Co-authored-by: Carlo Bramini <...>

### CODE REVIEW

**Review by HBelusca — APPROVED** [review]

**Comment by learn-more** [comment]
So @tkreuzer assigns this issue to himself, and that is a queue for you to send a part from a patch from 2016?

File: win32ss/gdi/eng/alphablend.c [hunk 1]
```diff
@@ -111,6 +111,7 @@ EngAlphaBlend(
 
     if (!IntEngEnter(&EnterLeaveDest, psoDest, &OutputRect, FALSE, &Translate, &OutputObj))
     {
+        IntEngLeave(&EnterLeaveSource);
```
Comment (tkreuzer) [inline win32ss/gdi/eng/alphablend.c:114]: No. When tha call to `IntEngEnter` fails, there is no reason to clean it up.

**Comment by SergeGautherie** [comment]
> So @tkreuzer assigns this issue to himself, and that is a queue for you to send a part from a patch from 2016?

I am not sure to understand what you are asking for. Timo did not do that. What is wrong with this PR?

**Comment by learn-more** [comment]
> > So @tkreuzer assigns this issue to himself, and that is a queue for you to send a part from a patch from 2016?
> 
> I am not sure to understand what you are asking for. Timo did not do that. What is wrong with this PR?

My mistake, I thought he assigned the ticket to himself recently, but he only changed the priority.

**Comment by binarymaster** [comment]
@tkreuzer can you please check whether this is correct? If yes, please submit an approval.

The patch looks trivial.

**Review by learn-more — APPROVED** [review]

**Pull request merged**